node_modules/col-assets/lib/api.js

```diff
@@ -281,133 +281,126 @@ var getAssets = module.exports.getAssets = function(ctx, filters, limit, offset,
     return callback({'code': 400, 'msg': validationResult.error.details[0].message});
   }
 
-  // Get the assets from the DB
-  var where = {
-    '$and': [
-      {
-        'course_id': ctx.course.id,
-        'visible': true
-      }
-    ]
+  // Get the assets from the DB. Because of the large number of (optional) includes,
+  // Sequelize doesn't appear to provide a way in which to combine this all, other
+  // than through a raw SQL query
+  var params = {
+    'course_id': ctx.course.id,
+    'user_id': ctx.user.id,
+    'types': filters.types,
+    'assignment': filters.assignment,
+    'category': filters.category,
+    'user': filters.user,
+    'limit': limit,
+    'offset': offset
   };
+
+  var sqlQuery = 'FROM assets a ' +
+                 'LEFT JOIN assets_categories ac ON a.id = ac.asset_id ' +
+                 'LEFT JOIN categories c ON c.id = ac.category_id ' +
+                 'LEFT JOIN asset_users au ON a.id = au.asset_id ' +
+                 'LEFT JOIN activities act ON a.id = act.asset_id' +
+                 ' AND act.course_id = :course_id' +
+                 ' AND act.user_id = :user_id' +
+                 ' AND act.object_type = \'asset\'' +
+                 ' AND act.type IN(\'like\', \'dislike\') ' +
+                 'WHERE a.deleted_at IS NULL' +
+                 ' AND a.course_id = :course_id ' +
+                 ' AND a.visible = TRUE ' +
+                 ' AND (c.visible = TRUE OR c.visible IS NULL)';
+
   if (filters.keywords) {
     // Replace spaces with wildcards so basic multi-term matching can happen
-    var keywords = filters.keywords.trim().replace(/ /g, '%');
-    where.$and.push({
-      '$or': {
-        'title': {
-          '$iLike': '%' + keywords + '%'
-        },
-        'description': {
-          '$iLike': '%' + keywords + '%'
-        }
-      }
-    });
+    var keywords = '%' + filters.keywords.trim().replace(/ /g, '%') + '%';
+    sqlQuery += ' AND (a.title ILIKE :keywords OR a.description ILIKE :keywords)';
+    params['keywords'] = keywords;
```

Comment:
@nicolaasmatthijs a little streamlining is possible:
`params['keywords'] =  '%' + filters.keywords.trim().replace(/ /g, '%') + '%';`
No need to declare `var keywords`, right?